provider/src/main/kotlin/org/gradle/kotlin/dsl/KotlinInitScript.kt

```diff
@@ -0,0 +1,384 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl
+
+import org.gradle.api.Action
+import org.gradle.api.PathValidation
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.ConfigurableFileTree
+import org.gradle.api.file.CopySpec
+import org.gradle.api.file.DeleteSpec
+import org.gradle.api.file.FileTree
+import org.gradle.api.invocation.Gradle
+import org.gradle.api.logging.Logger
+import org.gradle.api.logging.Logging
+import org.gradle.api.logging.LoggingManager
+import org.gradle.api.plugins.ObjectConfigurationAction
+import org.gradle.api.plugins.PluginAware
+import org.gradle.api.resources.ResourceHandler
+import org.gradle.api.tasks.WorkResult
+
+import org.gradle.kotlin.dsl.resolver.KotlinBuildScriptDependenciesResolver
+import org.gradle.kotlin.dsl.support.KotlinScriptHost
+import org.gradle.kotlin.dsl.support.serviceOf
+
+import org.gradle.process.ExecResult
+import org.gradle.process.ExecSpec
+import org.gradle.process.JavaExecSpec
+
+import java.io.File
+import java.net.URI
+
+import kotlin.script.extensions.SamWithReceiverAnnotations
+import kotlin.script.templates.ScriptTemplateDefinition
+
+
+/**
+ * Base class for Kotlin init scripts.
+ */
+@ScriptTemplateDefinition(
+    resolver = KotlinBuildScriptDependenciesResolver::class,
+    scriptFilePattern = ".+\\.init\\.gradle\\.kts")
+@SamWithReceiverAnnotations("org.gradle.api.HasImplicitReceiver")
+abstract class KotlinInitScript(
+    private val host: KotlinScriptHost,
+    gradle: Gradle) : Gradle by gradle {
+
+    /**
+     * Configures the classpath of the init script.
+     */
+    @Suppress("unused")
+    open fun initscript(@Suppress("unused_parameter") block: ScriptHandlerScope.() -> Unit) = Unit
+
+    /**
+     * Logger for settings. You can use this in your settings file to write log messages.
+     */
+    @Suppress("unused")
+    val logger: Logger by lazy { Logging.getLogger(Gradle::class.java) }
+
+    /**
+     * The [LoggingManager] which can be used to receive logging and to control the standard output/error capture for
+     * this script. By default, `System.out` is redirected to the Gradle logging system at the `QUIET` log level,
+     * and `System.err` is redirected at the `ERROR` log level.
+     */
+    @Suppress("unused")
+    val logging by lazy { gradle.serviceOf<LoggingManager>() }
+
+    /**
+     * Provides access to resource-specific utility methods, for example factory methods that create various resources.
+     */
+    @Suppress("unused")
+    val resources: ResourceHandler by lazy { fileOperations.resources }
+
+    /**
+     * Returns the relative path from this script's target base directory to the given path.
+     *
+     * The given path object is (logically) resolved as described for [KotlinSettingsScript.file],
```

Comment:
:x: `s/KotlinSettingsScript/KotlinInitScript/` and lots of other places below

Reply:
Right.